sysid/src/test/native/cpp/analysis/FilterTest.cpp

```diff
@@ -43,25 +43,96 @@ TEST(FilterTest, NoiseFloor) {
   EXPECT_NEAR(0.953, noiseFloor, 0.001);
 }
 
+void FillStepVoltageData(std::vector<sysid::PreparedData>& data) {
+  auto previousDatum = data.front();
+  for (int i = 1; i < data.size(); i++) {
+    auto& datum = data.at(i);
```

Comment:
```suggestion
  for (auto& datum : data) {
```

Reply:
I meant to start at index 1 for the data to not overwrite the first sample, but is that not needed here?